Tests/Tests/CustomEventSpec.swift

```diff
@@ -1,48 +1,57 @@
+                    expect(customEvent.analyticsParameters).to(beAKindOf([String: Any].self))
                 }
             }
+
             context("CustomEvent.getAttributeMap") {
-                it("will return non nil dictionary value") {
+                it("will return non nil dictionary value with CustomAttribute values") {
                     customEvent.customAttributes = customAttributes
                     expect(customEvent.getAttributeMap()).toNot(beNil())
+                    expect(customEvent.getAttributeMap()).to(beAKindOf([String: CustomAttribute]?.self))
                 }
-                it("will return nil value") {
+                it("will return nil value when customAttributes values are nil") {
                     customEvent.customAttributes = nil
                     expect(customEvent.getAttributeMap()).to(beNil())
                 }
             }
+
             context("CustomEvent.isEqual") {
                 it("will return false since the object is not of type Event") {
                     let commonUtility = CommonUtility()
                     expect(customEvent.isEqual(commonUtility)).to(beFalse())
                 }
-                it("will return false since the type and name of the object is different") {
-                    let newEvent = CustomEvent(withName: "Custom", withCustomAttributes: customAttributes)
-                    expect(customEvent.isEqual(newEvent)).to(beFalse())
+                it("will return false since name of the objects are different") {
+                    let event = CustomEvent(withName: "Custom", withCustomAttributes: customAttributes)
+                    expect(customEvent.isEqual(event)).to(beFalse())
                 }
-                it("will return true since the type and name is equal") {
-                    let newEvent2 = CustomEvent(withName: "Test4", withCustomAttributes: customAttributes)
-                    expect(customEvent.name.isEqual(newEvent2.name)).to(beTrue())
+                it("will return false since type and name of the object is different") {
+                    let event = Event(type: .purchaseSuccessful, name: "PurchaseSuccessful")
+                    expect(customEvent.isEqual(event)).to(beFalse())
                 }
-            }
-            context("CustomEvent.init") {
-                it("will return false since the super object is not equal") {
-                let custom = CustomEvent.init(withName: "customEvent", withCustomAttributes: customAttributes, timestamp: 60)
-                    expect(custom.timestamp).to(equal(60))
+                it("will return false since the attributes are different") {
+                    let customAttributes =  [CustomAttribute(withKeyName: "customKey1", withBoolValue: true),
+                                             CustomAttribute(withKeyName: "customkey2", withDoubleValue: 5)]
+                    let event = CustomEvent(withName: "Test", withCustomAttributes: customAttributes)
```

Comment:
Can you put `customEvent.name` instead of `"Test"` to ensure that the name will always be the same?